api/controllers/v2/github.ts

```diff
@@ -79,6 +133,139 @@ export const TriggerBuild = async (event: APIGatewayEvent): Promise<APIGatewayPr
     };
   }
 
+  if (!validateJsonWebhook(event, c.get<string>('githubSecret'))) {
+    const errMsg = "X-Hub-Signature incorrect. Github webhook token doesn't match";
+    return {
+      statusCode: 401,
+      headers: { 'Content-Type': 'text/plain' },
+      body: errMsg,
+    };
+  }
+
+  let body: WorkflowRunCompletedEvent;
+  try {
+    body = JSON.parse(event.body) as WorkflowRunCompletedEvent;
+  } catch (e) {
+    console.log('[TriggerBuild]: ERROR! Could not parse event.body', e);
+    return {
+      statusCode: 502,
+      headers: { 'Content-Type': 'text/plain' },
+      body: ' ERROR! Could not parse event.body',
+    };
+  }
+
+  if (body.workflow_run.conclusion != 'success')
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body: `Build on branch
+        ${body.workflow_run.head_branch}
+         is not complete and will not trigger smoke test site deployments`,
+    };
+
+  if (body.workflow_run.name != 'Deploy Staging ECS')
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body: `Workflow
+        ${body.workflow_run.name} completed successfully but only Deploy Staging ECS workflow completion will trigger smoke test site deployments`,
+    };
+
+  // if the build was not building main branch, no need for smoke test sites
+  // if (body.workflow_run.head_branch != 'main' || body.repository.fork) {
+  //   console.log('Build was not on master branch in main repo, sites will not deploy as no smoke tests are needed');
+  //   return {
+  //     statusCode: 202,
+  //     headers: { 'Content-Type': 'text/plain' },
+  //     body:
+  //       `Build on branch
+  //       ${body.workflow_run.head_branch}
+  //       will not trigger site deployments as it was not on main branch in upstream repo`,
+  //   };
+  // }
+
```

Comment:
Should this be removed or uncommented?

Reply:
This will be uncommented before merge, it just needs to be commented out so I can continue testing any changes